apis/r/R/ReadIter.R

```diff
@@ -28,14 +28,17 @@ ReadIter <- R6::R6Class(
     #' If read is complete, retunrs `NULL` and raises warning.
     #' @return \code{NULL} or one of arrow::\link[arrow]{Table}, \link{matrixZeroBasedView}
     read_next = function() {
-      if (is.null(private$soma_reader_pointer)) {
+      if (is.null(private$soma_reader_pointer)){
         return(NULL)
       }
-      if (self$read_complete()) {
+
+      tbl <- sr_next(private$soma_reader_pointer)
+
+      if(is.null(tbl)){
```

Comment:
Is there a reason we're getting rid of the `$read_complete()` logic?

Reply:
I am going to try reverting it back to what it use to be. It was breaking initially, but I think that's because I needed to update `sr_complete` in Rcpp.